planner/languages/csharp/csharp_planner.go

```diff
@@ -17,9 +29,88 @@ func (p *Planner) Name() string {
 }
 
 func (p *Planner) IsRelevant(srcDir string) bool {
-	return false
+	a, err := plansdk.NewAnalyzer(srcDir)
+	if err != nil {
+		// We should log that an error has occurred.
+		return false
+	}
+	return a.HasAnyFile("*.csproj")
 }
 
 func (p *Planner) GetPlan(srcDir string) *plansdk.Plan {
-	return &plansdk.Plan{}
+	plan, err := p.getPlan(srcDir)
+	if err != nil {
+		// Added this Printf because `devbox shell` was silently swallowing this error.
+		// TODO savil. Have `devbox shell` error out or print it instead.
+		fmt.Printf("error in getPlan: %s\n", err)
+		plan = &plansdk.Plan{}
+		plan.WithError(err)
+	}
+	return plan
+}
+
+func (p *Planner) getPlan(srcDir string) (*plansdk.Plan, error) {
+
+	proj, err := project(srcDir)
+	if err != nil {
+		return nil, err
+	}
+	dotNetPkg, err := dotNetNixPackage(proj)
+	if err != nil {
+		return nil, err
+	}
+
+	return &plansdk.Plan{
+		DevPackages: []string{dotNetPkg},
+	}, nil
+}
+
+func project(srcDir string) (*Project, error) {
+	a, err := plansdk.NewAnalyzer(srcDir)
+	if err != nil {
+		// We should log that an error has occurred.
+		return nil, err
+	}
+	paths := a.GlobFiles("*.csproj")
+	if len(paths) < 1 {
+		return nil, errors.Errorf("expected to find a .csproj file in directory %s", srcDir)
+	}
+	projectFilePath := paths[0]
+
+	content, err := os.ReadFile(projectFilePath)
+	if err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	proj := &Project{}
+	err = cuecfg.Unmarshal(content, ".xml", &proj)
+	if err != nil {
+		return nil, err
+	}
+
+	return proj, nil
+}
+
+// The TargetFramework is more complicated than below, but I'm picking out what
+// seem to be the common uses.
+// https://docs.microsoft.com/en-us/dotnet/standard/frameworks
+func dotNetNixPackage(proj *Project) (string, error) {
```

Comment:
Wondering if you can add `testdata/` for all the cases here.

I think you're missing:
- A project with a missing target framework
- donet 7
- dotnet 3
- Unknown/unrecognized version

Reply:
Added these cases except dotnet-7

I couldn't make a project for dotnet-7, because its nixpkg has been [published on 2022-08-22](https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.dotnet-sdk_7.x86_64-linux/all).

But we've pinned our nixpkgs to that of 2022-08-16
ref. from `shell.nix.tmpl` file:
```
  pkgs = import (fetchTarball {
    # Commit hash as of 2022-08-16
    # `git ls-remote https://github.com/nixos/nixpkgs nixos-unstable`
    url = "https://github.com/nixos/nixpkgs/archive/af9e00071d0971eb292fd5abef334e66eda3cb69.tar.gz";
    sha256 = "1mdwy0419m5i9ss6s5frbhgzgyccbwycxm5nal40c8486bai0hwy";
  }) {};
```